business/noticeService.js

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation and others. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+const { get } = require('lodash')
+const DocBuilder = require('tiny-attribution-generator/lib/docbuilder').default
+const TextRenderer = require('tiny-attribution-generator/lib/outputs/text').default
+const TemplateRenderer = require('tiny-attribution-generator/lib/outputs/template').default
+const JsonSource = require('tiny-attribution-generator/lib/inputs/json').default
+
+const logger = require('../providers/logging/logger')
+
+class NoticeService {
+  constructor(definitionService, attachmentStore) {
+    this.definitionService = definitionService
+    this.attachmentStore = attachmentStore
+    this.logger = logger()
+  }
+
+  async generate(coordinates, template = '') {
+    const renderer = template ? new TemplateRenderer(template) : new TextRenderer()
```

Comment:
question: does the template say what kind of renderer to use? For example, what if we want to use the JSON renderer? Feels like we should be saying what kind of render and then providing parameters to that. This would also apply at the REST API level

Reply:
let's implement all of chive and it's options